Components/Optimizers/StandardStochasticGradientDescent/itkStochasticGradientDescentOptimizer.h

```diff
@@ -177,23 +177,22 @@ class StochasticGradientDescentOptimizer : public ScaledSingleValuedNonLinearOpt
   void
   PrintSelf(std::ostream & os, Indent indent) const override;
 
-  /** Typedefs for multi-threading. */
-  using ThreaderType = itk::PlatformMultiThreader;
-  using ThreadInfoType = ThreaderType::WorkUnitInfo;
+  /** Typedef for multi-threading. */
+  using ThreadInfoType = MultiThreaderBase::WorkUnitInfo;
 
   // made protected so subclass can access
-  double                m_Value{ 0.0 };
-  DerivativeType        m_Gradient{};
-  ParametersType        m_SearchDir{};
-  ParametersType        m_PreviousSearchDir{};
-  ParametersType        m_PrePreviousSearchDir{};
-  ParametersType        m_MeanSearchDir{};
-  double                m_LearningRate{ 1.0 };
-  StopConditionType     m_StopCondition{ MaximumNumberOfIterations };
-  DerivativeType        m_PreviousGradient{};
-  DerivativeType        m_PrePreviousGradient{};
-  ParametersType        m_PreviousPosition{};
-  ThreaderType::Pointer m_Threader{ ThreaderType::New() };
+  double                     m_Value{ 0.0 };
+  DerivativeType             m_Gradient{};
+  ParametersType             m_SearchDir{};
+  ParametersType             m_PreviousSearchDir{};
+  ParametersType             m_PrePreviousSearchDir{};
+  ParametersType             m_MeanSearchDir{};
+  double                     m_LearningRate{ 1.0 };
+  StopConditionType          m_StopCondition{ MaximumNumberOfIterations };
+  DerivativeType             m_PreviousGradient{};
+  DerivativeType             m_PrePreviousGradient{};
+  ParametersType             m_PreviousPosition{};
+  MultiThreaderBase::Pointer m_Threader{ MultiThreaderBase::New() };
```

Comment:
Might use thread pools, so might _again_ improve the performance of StochasticGradientDescentOptimizer 👍